File: poo/2018362921_Tarea_03/tarea-03/src/main/java/mx/ipn/upiicsa/poo/OperationModes.java
package mx.ipn.upiicsa.poo;

import lombok.Getter;

@Getter
public enum OperationModes {
    PARAMETROS(1, "Parametros", "-p"),
    MENU(2, "Menu", "-m");

    private int id;
    private String nombre;
    private String acronimo;

    private OperationModes(int id, String nombre, String acronimo) {
        this.id = id;
        this.nombre = nombre;
        this.acronimo = acronimo;
    }

    public static OperationModes getOperationMode(String operationMode) throws RuntimeException {
        OperationModes[] operationModes = values();
        for (OperationModes operation : operationModes) {
            if (operation.getAcronimo().equals(operationMode)) {
                return operation;
            }
        }
        throw new RuntimeException("Modo de Operacion NO Soportado");
    }
}

File: poo/2018362921_Tarea_03/version_python/src/main/python/operation_modes.py
from enum import StrEnum

class OperationModes(StrEnum):
    MENU = "-m"
    PARAMETERS = "-p"
    
    def get_operation_mode(self, flag):
        for mode in OperationModes:
            if flag.lower().__eq__(mode.value):
                return mode;
        raise RuntimeError("Modo de Operacion no soportado") 